pulsar-broker/src/main/java/org/apache/pulsar/broker/transaction/buffer/impl/TopicTransactionBuffer.java

```diff
@@ -178,8 +178,9 @@ public void handleTxnEntry(Entry entry) {
                     }
 
                     @Override
-                    public void recoverExceptionally(Exception e) {
-                        if (e instanceof PulsarClientException.BrokerMetadataException) {
+                    public void recoverExceptionally(Throwable e) {
+                        if (e instanceof PulsarClientException.BrokerMetadataException
+                                || e instanceof PulsarClientException.BrokerPersistenceException) {
```

Comment:
Does the BrokerPersistenceException is introduced by the transaction buffer recovery stage? I think it should be introduced in adding entries to the transaction buffer?